test/e2e/operator.go

```diff
@@ -314,10 +312,19 @@ var _ = XDescribe("ARO Operator - MachineSet Controller", func() {
 		}
 		Expect(replicaCount).To(BeEquivalentTo(minSupportedReplicas))
 
-		// Restore previous state
+		// Scale back to previous state
 		for _, ms := range mss.Items {
-			err := scale(ms.Name, *ms.Spec.Replicas)
+			err = scale(ms.Name, *ms.Spec.Replicas)
 			Expect(err).NotTo(HaveOccurred())
 		}
+
+		for _, ms := range mss.Items {
+			err = waitForScale(ms.Name)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		// Wait for old machine objects to delete
+		err = waitForMachines()
```

Comment:
This part here confuses me. The comment says it is waiting for machines to delete, but the code waits for machines to be running.

How it is supposed to work? What it is waiting for?

Reply:
Machine objects have a few different states - Provisioned, Deleting, and Running - so if all machines are Running, we know none are Provisioned or Deleting, and all have joined the cluster as nodes.